vendor/views/vendor.py

```diff
@@ -332,7 +324,7 @@ class OrderHistoryListView(LoginRequiredMixin, ListView):
     def get_queryset(self):
         try:
             # The profile and user are site specific so this should only return what's on the site for that user excluding the cart
-            return self.request.user.customer_profile.get(site=get_current_site(self.request)).invoices.filter(status__gt=Invoice.InvoiceStatus.CART)
+            return self.request.user.customer_profile.get(site=self.request.site).invoices.filter(status__gt=Invoice.InvoiceStatus.CART)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(self.request)`